### ISSUE TITLE
Add HOC to add <Badge /> to an <Icon />

### ISSUE THREAD

**Issue by janhesters** [react-native-training/react-native-elements#1603]
**Is your feature request related to a problem? Please Describe.**
On mobile apps, Icons are often used in conjunction with badges. I often find myself adding badges to icons manually.

**Describe the solution you'd like**
I created a handy HOC for this purpose for myself. I could make a PR adding it to the code base. The result looks like this:
![simulator screen shot - iphone xs - 2018-11-26 at 14 27 14](https://user-images.githubusercontent.com/31096420/49017127-57760a00-f188-11e8-8139-f620ca6531f0.png)

**Describe alternatives you've considered**
An alternative would be to just add a recipe how to add the `<Badge />` to the `<Icon />` manually in the recipes section (see below). I can gladly write up the documentation for this.

**Additional context**
Some like this has been proposed in [this list of recipes](https://github.com/react-native-training/react-native-elements/issues/681).

**Comment by iRoachie** [comment]
I really like this idea~! Also if that's your design up there, it looks really slick 💯. How would the HOC work?

I don't usually like using HOCs but i'm open to suggestions

**Comment by janhesters** [comment]
The API looks like this:
```ts
withBadge(<Icon type="ionicon" name={getIconName("cart")} containerStyle={styles.iconRight} onPress={() => {}} />)(1)
```
But it can take in any component.

I created a PR so you can look at the code [here](https://github.com/react-native-training/react-native-elements/pull/1604).

**Comment by iRoachie** [comment]
Implemented in #1604
